packages/select/src/components/select/select2.tsx

```diff
@@ -251,6 +250,24 @@ export class Select2<T> extends AbstractPureComponent2<Select2Props<T>, Select2S
         ) : undefined;
     }
 
+    private withPopoverTargetPropsHandler = (
+        eventType: "keydown" | "keyup",
+        handler: React.KeyboardEventHandler<HTMLElement> | undefined,
+    ): React.KeyboardEventHandler<HTMLElement> => {
+        switch (eventType) {
+            case "keydown":
+                return event => {
+                    handler?.(event);
+                    this.props.popoverTargetProps?.onKeyDown?.(event);
```

Comment:
this is a bugfix -- `popoverTargetProps.onKey{Down|Up}` handlers were never getting called before.